detection_rules/integrations.py

```diff
@@ -106,3 +222,64 @@ def get_integration_manifests(integration: str) -> list:
     print(f"loaded {integration} manifests from the following package versions: "
           f"{[manifest['version'] for manifest in manifests]}")
     return manifests
+
+
+def get_integration_schema_data(data, meta) -> Generator[dict, None, None]:
+    """Iterates over specified integrations from package-storage and combines schemas per version."""
+
+    # lazy import to avoid circular import
+    from .rule import (  # pylint: disable=import-outside-toplevel
+        QueryRuleData, RuleMeta, TOMLRuleContents)
+
+    data: QueryRuleData
+    meta: RuleMeta
+
+    packages_manifest = load_integrations_manifests()
+    integrations_schemas = load_integrations_schemas()
+
+    # validate the query against related integration fields
+    if isinstance(data, QueryRuleData) and data.language != 'lucene':
+        package_integrations = TOMLRuleContents.get_packaged_integrations(data, meta, packages_manifest)
+
+        if not package_integrations:
+            return
+
+        for stack_version, mapping in meta.get_validation_stack_versions().items():
+            ecs_version = mapping['ecs']
+
+            ecs_schema = ecs.flatten_multi_fields(ecs.get_schema(ecs_version, name='ecs_flat'))
+
+            for pk_int in package_integrations:
+                package = pk_int["package"]
+                integration = pk_int["integration"]
+
+                if meta.min_stack_version is None and meta.maturity == "development":
```

Comment:
no min stack _and_ dev?

Reply:
We could skip on both conditions. I initially thought we min_stacked all rules but there are a few lingering that are not min_stacked. In particular, rules still in `development`.